R/util_dfc.R

```diff
@@ -0,0 +1,238 @@
+##########################################################################
+###
+###     util_dfc.R
+###     Dan Greene, 12/21/17
+###
+###     The goal of the util_dfc ("data frame compare") module is to make it easy to see
+###     where two data frames differ for debugging and further analysis.
+###
+###     Depends on util.R
+###
+
+util_dfc.compare_vecs <- function(vec1, vec2) {
+  # compare vector elements without regard to order and without checking for duplicates.
+  # return a list with three components -
+  ## shared_elements: the shared elements,
+  ## only_in_first: the elements in vec1 and not vec2,
+  ## only_in_second: the elements in vec2 and not vec1.
+  return(list(shared_elements = intersect(vec1, vec2),
+              only_in_first = setdiff(vec1, vec2),
+              only_in_second = setdiff(vec2, vec1)))
```

Comment:
I really like this only_in_first, only_in_second convention that you use throughout.